core/src/glue.rs

```diff
@@ -24,22 +24,49 @@ impl<T: GStore + GStoreMut> Glue<T> {
         Self { storage }
     }
 
-    pub async fn plan<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Vec<Statement>> {
+    pub async fn plan_with_params<Sql, I, P>(
+        &mut self,
+        sql: Sql,
+        params: I,
+    ) -> Result<Vec<Statement>>
+    where
+        Sql: AsRef<str>,
+        I: IntoIterator<Item = P>,
+        P: IntoParamLiteral,
+    {
         let parsed = parse(sql)?;
+        let params: Vec<ParamLiteral> = params
+            .into_iter()
+            .map(|value| value.into_param_literal())
+            .collect::<Result<_, TranslateError>>()?;
         let storage = &self.storage;
         stream::iter(parsed)
-            .map(|p| translate(&p))
+            .map(|p| translate_with_params(&p, params.clone()))
             .then(|statement| async move { plan(storage, statement?).await })
             .try_collect()
             .await
     }
 
+    pub async fn plan<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Vec<Statement>> {
+        self.plan_with_params(sql, std::iter::empty::<ParamLiteral>())
+            .await
+    }
+
     pub async fn execute_stmt(&mut self, statement: &Statement) -> Result<Payload> {
         execute(&mut self.storage, statement).await
     }
 
-    pub async fn execute<Sql: AsRef<str>>(&mut self, sql: Sql) -> Result<Vec<Payload>> {
-        let statements = self.plan(sql).await?;
+    pub async fn execute_with_params<Sql, I, P>(
```

Comment:
It would be nice to add a simple docs include `# Errors` section. For example:
```rust
/// Executes the statements in the SQL string using the supplied parameters
/// and returns their payloads.
///
/// # Errors
///
/// Fails when planning or executing any statement fails, including
/// translation errors while binding parameters.
```

Reply:
Applied!